compliance/collection/nodeinventorizer/nodeinventory.go

```diff
@@ -108,6 +108,26 @@ func convertAndDedupRHELComponents(rc *database.RHELv2Components) []*scannerV1.R
 	return maps.Values(convertedComponents)
 }
 
-func makeComponentKey(component *scannerV1.RHELComponent) string {
+func convertExecutables(exe []*scannerV1.Executable) []*storage.NodeInventory_Components_RHELComponent_Executable {
+	arr := make([]*storage.NodeInventory_Components_RHELComponent_Executable, len(exe))
+	for i, executable := range exe {
+		arr[i] = &storage.NodeInventory_Components_RHELComponent_Executable{
+			Path:             executable.Path,
+			RequiredFeatures: nil,
+		}
+		if executable.GetRequiredFeatures() != nil {
+			arr[i].RequiredFeatures = make([]*storage.NodeInventory_Components_RHELComponent_Executable_FeatureNameVersion, len(executable.GetRequiredFeatures()))
+		}
+		for i2, fnv := range executable.GetRequiredFeatures() {
```

Comment:
nit: would it make sense to nest this for loop in the previous block after the array allocation ?

Reply:
dded in [fdbd8cd](https://github.com/stackrox/stackrox/pull/4392/commits/fdbd8cd232b3fa7ac139ce100ad7d65fb06e7793)